File: app/src/main/java/com/demo/justapp/exchanger/presentation/base/BaseFragment.java
package com.demo.justapp.exchanger.presentation.base;

import com.arellomobile.mvp.MvpAppCompatFragment;
import com.demo.justapp.exchanger.utils.ComponentUtils;

import dagger.internal.Preconditions;

/**
 * @author Sergey Rodionov
 */
public class BaseFragment extends MvpAppCompatFragment {

    protected <C> C getComponent(Class<C> componentType) {
        Preconditions.checkNotNull(getActivity());
        return ComponentUtils.getComponent(getActivity(), componentType);
    }

}

File: app/src/main/kotlin/com/demo/justapp/exchanger/presentation/base/BaseFragment.kt
package com.demo.justapp.exchanger.presentation.base

import com.arellomobile.mvp.MvpAppCompatFragment
import com.demo.justapp.exchanger.exts.ComponentUtils

/**
 * Базовый фрагмент для приложения.
 * <p>
 * Тут будет какая-то общая логика для всех фрагментов в приложении.
 *
 * @author Sergey Rodionov
 */
open class BaseFragment : MvpAppCompatFragment() {

    protected fun <C> getComponent(componentType: Class<C>): C {
        val activity = checkNotNull(activity)
        return ComponentUtils.getComponent(activity, componentType)
    }

}